File: banchus-common/src/main/java/pe/nanamochi/banchus/domain/enums/BeatmapWebRankedStatus.java
package pe.nanamochi.banchus.domain.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum BeatmapWebRankedStatus {
  NOT_SUBMITTED(-1),
  PENDING(0),
  UPDATE_AVAILABLE(1),
  RANKED(2),
  APPROVED(3),
  QUALIFIED(4),
  LOVED(5);

  private final int value;

  public static BeatmapWebRankedStatus fromValue(int value) {
    for (BeatmapWebRankedStatus status : values()) {
      if (status.value == value) {
        return status;
      }
    }
    return NOT_SUBMITTED;
  }

  public static int convertToWebStatus(BeatmapRankedStatus rankedStatus) {
    if (rankedStatus == null) return NOT_SUBMITTED.getValue();

    return switch (rankedStatus) {
      case GRAVEYARD, WIP, PENDING -> PENDING.getValue();
      case RANKED -> RANKED.getValue();
      case APPROVED -> APPROVED.getValue();
      case QUALIFIED -> QUALIFIED.getValue();
      case LOVED -> LOVED.getValue();
    };
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/beatmap/enums/BeatmapWebRankedStatus.kt
package pe.nanamochi.banchus.beatmap.enums

enum class BeatmapWebRankedStatus(val value: Int) {
    NOT_SUBMITTED(-1),
    PENDING(0),
    UPDATE_AVAILABLE(1),
    RANKED(2),
    APPROVED(3),
    QUALIFIED(4),
    LOVED(5);

    companion object {
        fun fromValue(value: Int): BeatmapWebRankedStatus {
            return entries.find { it.value == value } ?: NOT_SUBMITTED
        }

        fun convertToWebStatus(rankedStatus: BeatmapRankedStatus?): Int {
            if (rankedStatus == null) return NOT_SUBMITTED.value

            return when (rankedStatus) {
                BeatmapRankedStatus.GRAVEYARD,
                BeatmapRankedStatus.WIP,
                BeatmapRankedStatus.PENDING -> PENDING.value

                BeatmapRankedStatus.RANKED -> RANKED.value
                BeatmapRankedStatus.APPROVED -> APPROVED.value
                BeatmapRankedStatus.QUALIFIED -> QUALIFIED.value
                BeatmapRankedStatus.LOVED -> LOVED.value
            }
        }
    }
}
